src/parser.rs

```diff
@@ -3588,5 +3628,47 @@ mod test {
         } else {
           panic!("Failed to read fake env float");
         }
+
```

Comment:
Thanks for the feedback. Will break down the test to smaller tests.
And about the error handling do you think the parser should generate any error here? Isn't it better to return an empty string if the environment variable is not set? My thougths are we are matching the value if it exists through all parsers and if the first one match will be used as the result otherwise we return the empty string.